nxviz/io.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+from networkx import Graph, MultiGraph
+
+
+def graph_from_dataframe(
+        dataframe,
+        threshold_by_percent_unique=0.1,
+        threshold_by_count_unique=None,
+        node_id_columns=[],
+        node_property_columns=[],
+        edge_property_columns=[],
+        collapse_edges=True,
+        edge_agg_key='weight'):
+    '''
+    Build an undirected graph from a pandas dataframe.
+    This function attempts to infer which cells should become nodes
+    based on either:
+        a) what percentage of the column are unique values (defaults to 10%)
+        b) an explicit count of unique values
+            (i.e. any column with 10 unique columns or less)
+        c) an explicit list of column keys
+            (i.e. ['employee_id', 'location_code'])
+    '''
+    assert isinstance(dataframe, pd.DataFrame), \
+        "{} is not a pandas DataFrame".format(dataframe)
+
+    M = MultiGraph()
+
+    # if explicit specification of node_id_columns is provided, use those
+    if len(node_id_columns) > 0:
+        node_columns = node_id_columns
+    else:
+        # otherwise, compute with thresholds based on the dataframe
+        if(threshold_by_count_unique):
+            node_columns = sorted([
+                    col for col in dataframe.columns
+                    if dataframe[col].nunique() <= threshold_by_count_unique
+                ])
+        else:
+            node_columns = sorted([
+                    col for col in dataframe.columns if
+                    dataframe[col].nunique() / dataframe.shape[0] <= threshold_by_percent_unique  # NOQA to preserve meaningful variable names
+                ])
+
+    # use the unique values for each node column as node types
+    for node_type in node_columns:
+        M.add_nodes_from([(node, {'type': node_type})
```

Comment:
It may be helpful to ensure that users of `nxviz.io` understand that there is a 'type' keyword that will get added to each node in the NetworkX graph that is returned.